Issue300 fix debug spectrograms

towsey opened the pull request:
Original intention was to fix drawing of spectrograms. This included the drawing of acoustic events and the hit matrix overlap. Therefore did some tidying up of the AcousticEvent class. Also set FFT window to HANNING which was originally set as Issue #291.
Wrote unit tests for the drawing of Acoustic events and the merging of overlapping acoustic events.

Work for #300.

towsey commented:
"hit matrix overlap" should read "hit matrix overlay"

atruskie commented on tests/Acoustics.Test/AudioAnalysisTools/AcousticEvents/EventTests.cs:
```diff
@@ -0,0 +1,157 @@
+        }
+
+        [TestMethod]
+        public void TestSonogramWithEventsOverlay()
```
For this test, I'd suggest using the `GeneratedImageTest` base class I've set up.

Also, the previous comment about ceremony applies here

atruskie commented on tests/Acoustics.Test/AudioAnalysisTools/AcousticEvents/EventTests.cs:
```diff
@@ -0,0 +1,157 @@
+                ev.DrawEvent(substituteSonogram, framesPerSecond, freqBinWidth, height);
+            }
+
+            //substituteSonogram.Save("C:\\temp\\image.png");
```
This is a bad example to leave in. Remove or fix path. See `GeneratedImageTests` for example output save directory

atruskie commented on tests/Acoustics.Test/AudioAnalysisTools/StandardSpectrograms/SonogramTests.cs:
```diff
@@ -219,5 +221,47 @@ public void TestAnnotatedSonogramWithPlots()
         }
+
+        [TestMethod]
+        public void TestSonogramHitsOverlay()
```
Use the generated image test class

towsey commented:
Concerning requested changes:
1: I have reverted the names of temporal variables as requested.
2: Changed signatures of test methods to simplify.
3: Removed methods that were commented out.
4: I have not used the GeneratedImageTest base class because I do not understand how to make use of the examples made available.
5: I created a new name space for Event tests, because it seems sensible to have all test classes of events in the one (more general) name space.

atruskie commented:
@towsey - thanks.

RE 4.: Set up a call.
RE 5. Tests should *roughly* mirror source. The event statistics class is in `AudioAnalysisTools.EventStatistics` in the source so the test should be in `Acoustics.Test.AudioAnalysisTools.EventStatistics`. What made sense to you is actually less consistent.

towsey commented:
@atruskie 
Concerning point 5, location of the EventStatistics class, my suggestion is that the source should be altered so that both the EventBase class, the AcousticEvent class and the EventStatistics class are all included within the same namespace, e.g. Event namespace.

I will fix the Anti Aliasing issue and then suggest a call  concerning point 4.

towsey commented:
I have not been able to find a solution to the anti-aliasing problem.
Can you send me an example or can we have a call when you are ready?
